code/mob/living/critter/brullbar.dm

```diff
@@ -36,26 +72,192 @@
+			APPLY_ATOM_PROPERTY(src, PROP_MOB_STUN_RESIST_MAX, "brullbar", 20)
 
 	setup_healths()
-		add_hh_flesh(100, 0.85)
-		add_hh_flesh_burn(100, 1.4)
+		add_hh_flesh(src.health_brute, src.health_brute_vuln)
+		add_hh_flesh_burn(src.health_burn, src.health_burn_vuln)
 		add_health_holder(/datum/healthHolder/toxin)
-		add_health_holder(/datum/healthHolder/brain)
+
+	seek_target(var/range = 9)
+		. = list()
+		for (var/mob/living/C in hearers(range, src))
+			if (isdead(C)) continue
+			if (isintangible(C)) continue //don't attack what you can't touch
+			if (istype(C, /mob/living/critter/brullbar)) continue //don't kill other brullbars
+			if (ishuman(C))
+				var/mob/living/carbon/human/H = C
+				if(!is_king && iswerewolf(H))
+					src.visible_message("<span class='alert'><b>[src] backs away in fear!</b></span>")
+					step_away(src, H, 15)
+					src.set_dir(get_dir(src, H))
+					continue
+			. += C
+
+		if (length(.) && prob(10))
+			playsound(src.loc, 'sound/voice/animal/brullbar_roar.ogg', 75, 1)
+			src.visible_message("<span class='alert'><B>[src]</B> roars!</span>")
+
+	seek_scavenge_target(var/range = 9)
+		. = list()
+		for (var/mob/living/M in view(range, get_turf(src)))
+			if (!isdead(M)) continue // eat everything yum
+			if (ishuman(M))
+				var/mob/living/carbon/human/H = M
+				if (H.decomp_stage <= 3 && !H.bioHolder?.HasEffect("husk")) //is dead, isn't a skeleton, isn't a grody husk
+					. += M
+
+	critter_attack(var/mob/target)
+		var/datum/targetable/critter/frenzy = src.abilityHolder.getAbility(/datum/targetable/critter/frenzy)
+		var/datum/targetable/critter/tackle = src.abilityHolder.getAbility(/datum/targetable/critter/tackle)
+		if (!tackle.disabled && tackle.cooldowncheck() && prob(20) && !is_incapacitated(target))
+			tackle.handleCast(target) // no return to wack people with the frenzy after the tackle sometimes
+		if (!frenzy.disabled && frenzy.cooldowncheck() && prob(40))
+			frenzy.handleCast(target)
+			return
+		else if (issilicon(target) && !ON_COOLDOWN(src, "brullbar_messup_silicon", 30 SECONDS))
+			fuck_up_silicons(target)
+			return
+		else
+			return ..()
+
+	critter_scavenge(var/mob/target)
+		if (prob(30))
+			src.visible_message("<span class='alert'><b>[src] devours [target]! Holy shit!</b></span>")
+			playsound(src.loc, 'sound/impact_sounds/Flesh_Break_1.ogg', 50, 1)
+			if (ishuman(target)) new /obj/decal/fakeobjects/skeleton(target.loc)
+			target.ghostize()
+			target.gib()
+			return
+		else
+			src.visible_message("<span class='alert'<b>[src] bites a chunk out of [target]!</b></span>")
+			playsound(src.loc, 'sound/items/eatfood.ogg', 30, 1)
+			for(var/damage_type in src.healthlist)
+				var/datum/healthHolder/hh = src.healthlist[damage_type]
+				hh.HealDamage(10)
+			return
+
+	can_critter_attack()
+		var/datum/targetable/critter/frenzy = src.abilityHolder.getAbility(/datum/targetable/critter/frenzy)
+		return can_act(src,TRUE) && !frenzy.disabled
```

Comment:
This seems a little odd - why are you only checking if one ability is disabled? It seems like this should just the parent, since the brullbar can attack with either ability or with limbs.

Reply:
its so it can't attack you while its using frenzy

Reply:
but probably not good to do it like this

Reply:
actually no this should be right

Reply:
that makes sense. Might be worth adding a comment so that that's more clear, but otherwise it checks out.